### PR TITLE
activate mail sending 

### LINKED ISSUE #31: Add Email Functionality
Add email functionality removed by #30

### PR DESCRIPTION
**Pull request by atomfrede**
This PR uses http://www.simplejavamail.org to replace the spring java mailing integration.

closes #31

### CODE REVIEW

**Comment by JasonTypesCodes** [comment]
This is great @atomfrede !  It seems the tests aren't happy.

**Comment by atomfrede** [comment]
Thought I fixed everything. Will have a look today.

**Comment by atomfrede** [comment]
Will look into the failing oauth tests tomorrow. Guess it tries to send mails  although it shouldn't.

File: generators/server/templates/build.gradle.ejs [hunk 1]
```diff
@@ -287,6 +287,7 @@ dependencies {
         exclude group: "org.springframework.boot", module: "spring-boot-starter-aop"
         exclude group: "org.springframework.boot", module: "spring-boot-starter-web"
     }
+    implementation 'org.simplejavamail:simple-java-mail:6.4.3'
```
Comment (JasonTypesCodes) [inline generators/server/templates/build.gradle.ejs:290]: Can we pull this version into the constants file?

Comment (atomfrede) [reply generators/server/templates/build.gradle.ejs]: :heavy_check_mark:

File: generators/server/templates/build.gradle.ejs [hunk 2]
```diff
@@ -287,7 +287,7 @@ dependencies {
         exclude group: "org.springframework.boot", module: "spring-boot-starter-aop"
         exclude group: "org.springframework.boot", module: "spring-boot-starter-web"
     }
-    implementation 'org.simplejavamail:simple-java-mail:6.4.3'
+    implementation 'org.simplejavamail:simple-java-mail:$simplejavamail_version'
```
Comment (atomfrede) [inline generators/server/templates/build.gradle.ejs:290]:
```suggestion
    implementation "org.simplejavamail:simple-java-mail:$simplejavamail_version"
```

**Comment by atomfrede** [comment]
Bounty claimed https://opencollective.com/generator-jhipster/expenses/24951

**Comment by pascalgrimaud** [comment]
@atomfrede : approved

**Pull request merged**